RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -1260,25 +1318,104 @@ class DeepTauId : public deep_tau::DeepTauBase {
       return false;
   }
 
-  inline void checkInputs(
-      const tensorflow::Tensor& inputs, const char* block_name, int n_inputs, int n_eta = 1, int n_phi = 1) const {
+  inline void checkInputs(const tensorflow::Tensor& inputs, const std::string& block_name, int n_inputs, const CellGrid* grid = nullptr) const {
+    if (debug_level >= 1) {
+      std::cout << "<checkInputs>: block_name = " << block_name << std::endl;
+      if ( block_name == "input_tau" ) {
+        for (int input_index = 0; input_index < n_inputs; ++input_index) {
+          float input = inputs.matrix<float>()(0, input_index);
+          if (edm::isNotFinite(input)) {
+            throw cms::Exception("DeepTauId")
+              << "in the " << block_name << ", input is not finite, i.e. infinite or NaN, for input_index = " << input_index;
+          }
+          if (debug_level >= 2) {
+            std::cout << block_name << "[var = " << input_index << "] = " << std::setprecision(5) << std::fixed << input << std::endl;
+          }
+        }
+      } else {
+        assert(grid);
+        int n_eta, n_phi;
+        if ( block_name.find("input_inner") != std::string::npos ) {
+          n_eta =  5;
+          n_phi =  5;
+        } else if ( block_name.find("input_outer") != std::string::npos ) {  
+          n_eta = 10;
+          n_phi = 10;
+        } else assert(0);
+        int eta_phi_index = 0;
+        for (int eta = -n_eta; eta <= n_eta; ++eta) {
+          for (int phi = -n_phi; phi <= n_phi; ++phi) {
+            const CellIndex cell_index{eta, phi};
+            const auto cell_iter = grid->find(cell_index);
+            if (cell_iter != grid->end()) {
+              for (int input_index = 0; input_index < n_inputs; ++input_index) {
+                float input = inputs.tensor<float, 4>()(eta_phi_index, 0, 0, input_index);
+                if (edm::isNotFinite(input)) {
+                  throw cms::Exception("DeepTauId")
+                    << "in the " << block_name << ", input is not finite, i.e. infinite or NaN, for eta = " << eta
+                    << ", phi = " << phi << ", input_index = " << input_index;
+                }
+                if (debug_level >= 2) {
+                  std::cout << block_name << "[eta = " << eta << "][phi = " << phi << "][var = " << input_index << "] = " << std::setprecision(5) << std::fixed << input << std::endl;
+                }
+              }
+              eta_phi_index += 1;
+            }
+          }
+        }
+      }
+    }
+  }
+
+  inline void saveInputs(const tensorflow::Tensor& inputs, const std::string& block_name, int n_inputs, const CellGrid* grid = nullptr) const {
```

Comment:
Is it correct that you switched the grids from a ref type to a pointer type in order to be allowed to set the nullptr as default here? Is there another reason? (just for my understanding)

Reply:
Yes, the grid is passed by pointer so that it can be a nullptr. The nullptr is used in the call
  saveInputs(*tauBlockTensor_, "input_tau", dnn_inputs_2017_v2::TauBlockInputs::NumberOfInputs);
(L1619 in my version of the DeepTauId.cc code)